cleanlab/segmentation/summary.py

```diff
@@ -0,0 +1,354 @@
+
+    labels:
+      Optional discrete array of noisy labels for a segmantic segmentation dataset, in the shape``(N,H,W,)``.
+      where each pixel must be integer in 0, 1, ..., K-1.
+      If `labels` is provided, this function also displays given label of the pixel identified with issue.
+      Refer to documentation for this argument in :py:func:find_label_issues <cleanlab.segemntation.filter.find_label_issues> for more information.
+
+    pred_probs:
+      Optional array of shape ``(N,K,H,W,)`` of model-predicted class probabilities.
+      If `pred_probs` is provided, this function also displays predicted label of the pixel identified with issue.
+      Refer to documentation for this argument in :py:func:find_label_issues <cleanlab.segemntation.filter.find_label_issues> for more information.
+
+      Tip: If your labels are one hot encoded you can `np.argmax(labels_one_hot,axis=1)` assuming that `labels_one_hot` is of dimension (N,K,H,W)
+      before entering in the function
+
+    class_names:
+      Optional list of strings, where each string represents the name of a class in the semantic segmentation problem.
+      The order of the names should correspond to the numerical order of the classes. The list length should be
+      equal to the number of unique classes present in the labels.
+      If provided, this function will generate a legend
+      showing the color mapping of each class in the provided colormap.
+
+      Example:
+        If there are three classes in your labels, represented by 0, 1, 2, then class_names might look like this:
+        class_names = ['background', 'person', 'dog']
+
+    top:
+        Optional maximum number of issues to be printed. If not provided, a good default is used.
+
+    exclude:
+        Optional list of label classes that can be ignored in the errors, each element must be 0, 1, ..., K-1
+
+    """
+    class_names, exclude, top = _get_summary_optional_params(class_names, exclude, top)
+    if labels is None and len(exclude) > 0:
+        raise ValueError("Provide labels to allow class exclusion")
+
+    top = min(top, len(issues))
+
+    correct_ordering = np.argsort(-np.sum(issues, axis=(1, 2)))[:top]
+
+    try:
+        import matplotlib.pyplot as plt
+        import matplotlib.patches as mpatches
+    except:
+        raise ImportError('try "pip install matplotlib"')
+
+    output_plots = (pred_probs is not None) + (labels is not None) + 1
+
+    # Colormap for errors
+    error_cmap = plt.cm.colors.ListedColormap(["none", "red"])
+    _, h, w = issues.shape
+    if output_plots > 1:
+        if pred_probs is not None:
+            _, num_classes, _, _ = pred_probs.shape
+            cmap = _generate_colormap(num_classes)
+        elif labels is not None:
+            num_classes = max(np.unique(labels)) + 1
+            cmap = _generate_colormap(num_classes)
+        else:
+            cmap = None
```

Comment:
is it concerning this line is never tested?  If not, then just leave as is, hard for me to tell